File: app/lib/models/expense.dart
class ExpenseModel {
  final int? id;
  final int userId;
  final int categoryId;
  final int? paymentMethodId;
  final double amount;
  final String expenseDate;
  final String? note;
  final String? createdAt;
  final String? updatedAt;

  // Presentation fields
  final String? categoryName;
  final String? categoryIconKey;
  final String? categoryColorHex;
  final String? paymentMethodName;
  final bool isPendingSync;

  ExpenseModel({
    this.id,
    required this.userId,
    required this.categoryId,
    this.paymentMethodId,
    required this.amount,
    required this.expenseDate,
    this.note,
    this.createdAt,
    this.updatedAt,
    this.categoryName,
    this.categoryIconKey,
    this.categoryColorHex,
    this.paymentMethodName,
    this.isPendingSync = false,
  });

  factory ExpenseModel.fromJson(Map<String, dynamic> json) {
    return ExpenseModel(
      id: json['id'] as int?,
      userId: (json['userId'] as num?)?.toInt() ?? 1,
      categoryId: (json['categoryId'] as num).toInt(),
      paymentMethodId: (json['paymentMethodId'] as num?)?.toInt(),
      amount: (json['amount'] as num).toDouble(),
      expenseDate: json['expenseDate'] as String,
      note: json['note'] as String?,
      createdAt: json['createdAt'] as String?,
      updatedAt: json['updatedAt'] as String?,
      categoryName: json['categoryName'] as String?,
      categoryIconKey: json['categoryIconKey'] as String?,
      categoryColorHex: json['categoryColorHex'] as String?,
      paymentMethodName: json['paymentMethodName'] as String?,
      isPendingSync: json['isPendingSync'] == true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'id': id,
      'userId': userId,
      'categoryId': categoryId,
      if (paymentMethodId != null) 'paymentMethodId': paymentMethodId,
      'amount': amount,
      'expenseDate': expenseDate,
      if (note != null) 'note': note,
    };
  }
}

File: backend/src/main/java/com/expensetracker/model/Expense.java
package com.expensetracker.model;

import java.math.BigDecimal;
import java.sql.Date;
import java.sql.Timestamp;

public class Expense {
    private Long id;
    private Long userId;
    private Long categoryId;
    private Long paymentMethodId;
    private BigDecimal amount;
    private Date expenseDate;
    private String note;
    private Timestamp createdAt;
    private Timestamp updatedAt;

    // Joined fields for easy presentation
    private String categoryName;
    private String categoryIconKey;
    private String categoryColorHex;
    private String paymentMethodName;

    public Expense() {}

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public Long getUserId() { return userId; }
    public void setUserId(Long userId) { this.userId = userId; }

    public Long getCategoryId() { return categoryId; }
    public void setCategoryId(Long categoryId) { this.categoryId = categoryId; }

    public Long getPaymentMethodId() { return paymentMethodId; }
    public void setPaymentMethodId(Long paymentMethodId) { this.paymentMethodId = paymentMethodId; }

    public BigDecimal getAmount() { return amount; }
    public void setAmount(BigDecimal amount) { this.amount = amount; }

    public Date getExpenseDate() { return expenseDate; }
    public void setExpenseDate(Date expenseDate) { this.expenseDate = expenseDate; }

    public String getNote() { return note; }
    public void setNote(String note) { this.note = note; }

    public Timestamp getCreatedAt() { return createdAt; }
    public void setCreatedAt(Timestamp createdAt) { this.createdAt = createdAt; }

    public Timestamp getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(Timestamp updatedAt) { this.updatedAt = updatedAt; }

    public String getCategoryName() { return categoryName; }
    public void setCategoryName(String categoryName) { this.categoryName = categoryName; }

    public String getCategoryIconKey() { return categoryIconKey; }
    public void setCategoryIconKey(String categoryIconKey) { this.categoryIconKey = categoryIconKey; }

    public String getCategoryColorHex() { return categoryColorHex; }
    public void setCategoryColorHex(String categoryColorHex) { this.categoryColorHex = categoryColorHex; }

    public String getPaymentMethodName() { return paymentMethodName; }
    public void setPaymentMethodName(String paymentMethodName) { this.paymentMethodName = paymentMethodName; }
}
